orttraining/orttraining/training_api/checkpoint.cc

```diff
@@ -550,10 +552,20 @@ Status OrtLoadInternal(const PathString& checkpoint_path,
     const auto tensor_file_full_path = ConcatPathComponent<PathChar>(checkpoint_path, tensor_file_path);
     LoadTensorProtoFromFile(tensor_file_full_path, tensor_protos, "[params]");
 
-    for (const auto& tensor_proto : tensor_protos) {
-      param_tensor_protos.push_back(tensor_proto);
+    for (auto& tensor_proto : tensor_protos) {
+      param_tensor_protos.emplace(std::make_pair(tensor_proto.name(), std::move(tensor_proto)));
```

Comment:
```suggestion
      auto tensor_proto_name = tensor_proto.name();
      param_tensor_protos.emplace(std::move(tensor_proto_name), std::move(tensor_proto));
```
order of function argument evaluation is unspecified. the situation we want to avoid is calling `tensor_proto.name()` after `tensor_proto` has been moved from.
see https://en.cppreference.com/w/cpp/language/eval_order and https://stackoverflow.com/questions/15680489/move-semantics-argument-evaluation-order

Reply:
Thanks for the suggestion, added a commit to solve this.